bridge-to-the-moon/src/verificationGames/challenger.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * implements a state machine for carrying out the verification game
+ *   from the challenger's perspective
+ */
+
+const StateMachine = require('javascript-state-machine')
+const waitForEvent = require('../utils/waitForEvent')
+
+module.exports = (web3, api) => ({
+  run: async (cmd, claim, sessionId) => new Promise(async (resolve, reject) => {
+    try {
+      // onBefore, onEnter, onLeave, onAfter
+      const m = new StateMachine({
+        init: 'init',
+        transitions: [
+          { name: 'query', from: ['init', 'preRespond'], to: 'preRespond' },
+          { name: 'finish', from: ['preRespond'], to: 'preFinish' },
+          { name: 'didConvict', from: '*', to: 'convicted' },
+        ],
+        methods: {
+          onBeforeQuery: async (tsn) => {
+            cmd.log('Syncing Session State...')
+            const session = await api.getSession(claim.id)
+            cmd.log('Synced.')
+
+            // let's check the most recent submission
+            let newMedStep
+            if (session.medStep.eq(0)) {
+              // is this the first query? we don't have a submission to check
+              // so we'll just go in the middle
+              cmd.log('This is the first query()')
+              newMedStep = Math.floor((session.highStep - session.lowStep) / 2)
```

Comment:
Shouldn't this be `session.lowStep + Math.floor((session.highStep - session.lowStep) / 2)`?

Reply:
(this also applies to all the others below)

Reply:
@chriseth – what you're saying sounds right to me. @Shrugs – what do you think?

Reply:
Good catch, updateded in d01ef93